release/src/module_dependencies/main.py

```diff
@@ -0,0 +1,172 @@
+import urllib.request
+import json
+import base64
+import networkx as nx
+
+# Sorts the ballerina standard library module list in ascending order
+def sortModuleNameList():
+    with open('./release/resources/module_list.json') as f:
+          nameList = json.load(f)
+
+    nameList['modules'].sort()
+    
+    with open('./release/resources/module_list.json', 'w') as jsonFile:
+        jsonFile.seek(0) 
+        json.dump(nameList, jsonFile, indent=4)
+        jsonFile.truncate()
+        
+    return nameList['modules'] 
+
+# Gets dependencies of ballerina standard library module from build.gradle file in module repository
+# returns: list of dependencies
+def getDependencies( balModule ):
+
+    data = urllib.request.urlopen("https://raw.githubusercontent.com/ballerina-platform/" + balModule + "/master/build.gradle")
+
+    dependencies = []
+
+    for line in data:
+
+        processedLine = line.decode("utf-8")
+
+        if 'ballerina-platform/module' in processedLine:
+            module = processedLine.split('/')[-1]
+            if module[:-2] == balModule:
+                continue
+            dependencies.append(module[:-2])
+
+    return dependencies
+
+# Gets the version of the ballerina standard library module from gradle.properties file in module repository
+# returns: current version of the module
+def getVersion(balModule):
+
+    data = urllib.request.urlopen("https://raw.githubusercontent.com/ballerina-platform/" + balModule + "/master/gradle.properties")
+
+    for line in data:
+
+        processedLine = line.decode("utf-8")
+
+        if 'version' in processedLine:
+            version = processedLine.split('=')[-1][:-1]
+
+    return version 
+
+# access the module_list.json file
+# returns: a list containing an array of standard library module names
+def getModuleNameList():
+    with open('./release/resources/module_list.json') as f:
+          fileContent = json.load(f)
+
+    return fileContent['modules']
+
+# Calculates the longest path between source and destination modules and replaces dependents that have intermediates
+def removeModulesInIntermediatePaths(G, source, destination, successors, moduleDetailsJSON):
+
```

Comment:
Extra empty line? Check for other similar cases as well.

Reply:
Fixed